src/file.cpp

```diff
@@ -937,6 +937,20 @@ namespace libtorrent
 		branch.append(leaf.data(), leaf.size());
 	}
 
+	std::string combine_path(std::string const& ls, std::string const& rs)
+	{
+		string_view lhs{ls};
+		string_view rhs{rs};
+		return combine_path(lhs, rhs);
+	}
+
+	std::string combine_path(char const* ls, char const* rs)
```

Comment:
this overload will also be sub-optimal as constructing the string views will have to run ``strlen()``. The proper overload would be:

``template <std::size_t Len1, std::size_t len2> std::string combine_path(char const (&ls)[Len1], char const (&rs)[Len2]);``

But doing this for you is the whole point of string_view in the first place.